docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,290 @@
+.. Copyright (c) 2020 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+TypeScript!
+###########
+
+daml2ts : TypeScript of DAML
+============================
+
+Introduction
+------------
+
+``daml2ts`` generates TypeScript for interacting with DAML via the `json-api <../json-api/index.html>`_ .
+
+``daml2ts`` inputs are compiled DAML modules. ``daml2ts`` outputs are TypeScript source files containing definitions modeling the DAML entities found.
+
+The code ``daml2ts`` generates is in terms of the package `@daml/types <https://github.com/digital-asset/daml/tree/master/language-support/ts/daml-types>`_.
+
+Primitives: @daml/types
+-----------------------
+
+To understand the TypeScript code generated by ``daml2ts``, it is helpful to keep in mind this quick review of the primitives provided by @daml/types.
+
+**Interfaces**:
+
+- ``Serializable<T>``
+- ``Template<T extends object, K = unknown>``
+- ``Choice<T extends object, C, R, K = unknown>``
+
+  ..
+        An object that extends the ``Serializable<T>`` interface has a field ``decoder: () => jtv.Decoder<T>``.
+        An object that extends the ``Template<T, K>`` interface has fields ``templateId: string``, ``keyDecoder: ()=> jtv.Decoder<K>``, ``Archive: Choice<T, {}, {}>`` and extends ``Serializable<T>``.
+        An object that extends the ``Choice<T, C, R, K>`` interface has fields ``template: () => Template<T, K>``, ``argumentDecoder: () => jtv.Decoder<C>``, ``resultDecoder: () => jtv.Decoder<R>`` and ``choiceName: string``.
```

Comment:
Let's remove the decoder stuff from here. It's an implementation detail and nobody knows what `jtv` refers to anyway. An object "implements" an interface rather than "extends" it. I would say something like

> An object that implements the `Template<T, K>` interface provides metadata for a template of type `T` with a potential contract key of type `K`, where the absence of contract key is marked by `K = undefined`. Such metadata includes the name of the template, the module and package where it is defined, and a description of the types `T` and `K`.

Similarly for choices.

Reply:
Oops. Sorry not obvious (since you're reviewing the ReST), that section you are commenting on is a comment 😄 .

Reply:
✔️